### PR TITLE
Update typing-extensions to 4.6.1

### PR DESCRIPTION
**Pull request by adriangb**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -268,7 +268,7 @@ jobs:
       fail-fast: false
       matrix:
         # test the oldest supported version and main
-        typing-extensions-version: ['4.5.0', 'main']
+        typing-extensions-version: ['4.6.1', 'main']
```
Comment (hramezani) [inline .github/workflows/ci.yml:271]: 4.6.2 is out. Should we use that one?

Comment (adriangb) [reply .github/workflows/ci.yml:271]: No this should be the minimum supported version. We don't need 4.6.2 as far as I can tell so let's stick to 4.6.1 for now.

**Review by samuelcolvin — APPROVED** [review]

**Comment by fergusdixon** [comment]
A consequence of this is that Tensorflow is not compatible with Pydantic V2 due to the restraints on typing_extensions
What was the reasoning behind restricting this?

**Comment by adriangb** [comment]
We wanted to support TypedDict and TypeAliasType.

I'm not sure why TensorFlow puts an upper bound on te, it's a relatively stable package.

**Comment by fergusdixon** [comment]
Seems to be a lot of back and forth between TF, wrapt, and typing-extensions over who exactly needs to fix what
They're tracking it here https://github.com/tensorflow/tensorflow/issues/60687

**Comment by adriangb** [comment]
Thanks for the link. Not much we can do at this point, hopefully TF / CPython / TE solve it eventually.

**Pull request merged**